cmd/carbonapi/http/render_handler.go

```diff
@@ -332,11 +348,62 @@ func renderHandler(w http.ResponseWriter, r *http.Request) {
 
 	if len(results) != 0 {
 		tc := time.Now()
-		config.Config.QueryCache.Set(cacheKey, body, cacheTimeout)
+		config.Config.ResponseCache.Set(responseCacheKey, body, responseCacheTimeout)
 		td := time.Since(tc).Nanoseconds()
 		ApiMetrics.RenderCacheOverheadNS.Add(td)
 	}
 
 	gotErrors := len(errors) > 0
 	accessLogDetails.HaveNonFatalErrors = gotErrors
 }
+
+func backendCacheComputeKey(from string, until string, targets []string) string {
```

Comment:
you can combine from and until there. I mean instead of
```
func backendCacheComputeKey(from string, until string, targets []string) string {
```

you can write
```
func backendCacheComputeKey(from, until string, targets []string) string {
```

That's just to make the linter a bit happier.

Otherwise PR looks good to me on a first glance.

Meanwhile I'll read it more careful.

Reply:
Thanks, I amended the commit with a fix.